### ISSUE TITLE
Issue of MYSQL_OPT_SSL_MODE

### ISSUE THREAD

**Issue by wujysh** [akopytov/sysbench#287]
Currently there are many `#ifdef MYSQL_OPT_SSL_MODE` in `drv_mysql.c`.
https://github.com/akopytov/sysbench/blob/master/src/drivers/mysql/drv_mysql.c#L67

But where does this macro `MYSQL_OPT_SSL_MODE` be defined? In `mysql.h`, it is only a value of enum.
I don't think all compilers can recognize this.

In my test environment:
```
Ubuntu 18.04.1 LTS
gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
libmysqlclient-dev/bionic-updates,bionic-security,now 5.7.24-0ubuntu0.18.04.1 amd64 [installed]
```
The compilation won't go into the #ifdef blocks, and the sysbench built has SSL issues.
Also, the output of `sysbench --help` contains the wrong parameter:
`--mysql-ssl[=on|off]             use SSL connections, if available in the client library [off]`

**Comment by akopytov** [comment]
You are right. That `#ifdef` will not work with any compiler. I made wrong assumptions when implementing that code.

**Comment by wujysh** [comment]
Besides, the code inside `#ifdef` also has issue.
After removed the `#ifdef` blocks, the compilation aborted with error `'ssl_mode_typelib' undeclared`.
```
drv_mysql.c: In function ‘mysql_drv_init’:
drv_mysql.c:269:46: error: ‘ssl_mode_typelib’ undeclared (first use in this function); did you mean ‘ssl_mode_string’?
   args.ssl_mode = find_type(ssl_mode_string, ssl_mode_typelib, FIND_TYPE_BASIC);
                                              ^~~~~~~~~~~~~~~~
                                              ssl_mode_string
```

**Comment by akopytov** [comment]
This should be fixed in both 1.0 and master.

**Comment by wujysh** [comment]
Thanks for your work.
